Feature/mc 12506 updating api documentation for kubernetes storage entities (#227)
cloudops/cloudmc-api-docs: source/includes/kubernetes_extension/_k8_persistentvolumes.md

    "metadata": {
        "annotations": {
            "key": "value"
        },
        "name": "persistentvolumes-name"
    },
    "spec": {
        "accessModes": [
            "ReadWriteMany"
        ],
        "capacity": {
            "storage": "2Gi"
        },
        "claimRef": {
            "apiVersion": "v1",
            "kind": "PersistentVolumeClaim",
            "name": "pv-claim-name",
            "namespace": "default",
            "resourceVersion": "259087",
            "uid": "11957feb-1acc-4c7e-911e-f77d5d7be5ac"
        },
        "hostPath": {
            "path": "/etc/path"
        },
        "persistentVolumeReclaimPolicy": "Retain",
        "storageClassName": "standard",
        "volumeMode": "Filesystem"
    }
}
```

> The above command returns a JSON structured like this:

```json
{
  "taskId": "1542bd45-4732-419b-87b6-4ea6ec695c2b",
  "taskStatus": "PENDING"
}
```

<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumes/:id?cluster_id=a_cluster_id</code>

Replace a persistent volume in a given [environment](#administration-environments).

> Required Attributes                 | &nbsp;

Reviewer: I think we use `Attributes` as it implicitly means required attributes. Can we check with other sections of the API docs to make sure that it's consistent across the doc?

Author: Its Required attributes in other sections

Reviewer: Thanks for checking that.

----------------------------------- | ------------------------------------------------------------
`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
`metadata` <br/>_object_            | The metadata of the pod.
`metadata.name` <br/>_string_       | The name of the pod.

> `spec`<br/>_object_                 | The specification used to replace and run the pod.

Reviewer 2: pod -> persistent volume

`spec.accessModes`<br/>_string_ | The desired access modes the volume should have.
`spec.capacity.storage`<br/>_string_  | The size of the claim.
`spec.hostPath.path`<br/>_string_  | The path of the directory on the host.

| Optional Attributes                       | &nbsp;                                                                  |
| ----------------------------------------- | ----------------------------------------------------------------------- |

> | `metadata.annotations` <br/>_string_        | Annotations is an unstructured key value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.|

Reviewer: ```suggestion
| `metadata.annotations` <br/>_string_        | Annotations are unstructured key-value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.|
```

Author: I think the first makes for sense to me, Annotations is a key value map that holds multiple keys and values. Whereas annotations.<key> are unstructured key-value pairs.

Reviewer: Yes, that's true. I was thinking more in terms of describing the JSON format and fields in the request payload.
In code, we do store these key-value pairs in Map but in JSON we only have the concept of key-value. WYT?

| `spec.claimRef` <br/>_string_        | part of a bi-directional binding between PersistentVolume and PersistentVolumeClaim. Expected to be non-nil when bound.|
| `spec.persistentVolumeReclaimPolicy` <br/>_string_        | What happens to a persistent volume when released from its claim.|
| `spec.storageClassName` <br/>_string_        | Name of StorageClass to which this persistent volume belongs. Empty value means that this volume does not belong to any StorageClass.|
| `spec.volumeMode` <br/>_string_        | Defines if a volume is intended to be used with a formatted filesystem or to remain in raw block state. Value of Filesystem is implied when not included in spec.|

Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#persistentvolume-v1-core).

Return value:

| Attributes                 | &nbsp;                                       |
| -------------------------- | -------------------------------------------- |

> | `taskId` <br/>_string_     | The id corresponding to the replace pod task.|

Reviewer 2: pod -> persistent volume
